add new sso and scim faqs (#8963)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/manage-access/scim-faq.md

---
title: "SCIM FAQs and troubleshooting"
description: "Common questions and troubleshooting for SCIM provisioning in dbt platform"
id: "scim-faq"
sidebar: "SCIM FAQ and troubleshooting"
---

> # SCIM FAQ and troubleshooting <Lifecycle status="managed, managed_plus" />

Reviewer: ```suggestion
# SCIM FAQs and troubleshooting <Lifecycle status="managed, managed_plus" />
```

Find answers to common questions about configuring and using SCIM provisioning in <Constant name="dbt_platform" />, plus guidance for resolving common issues.

## FAQs

<Expandable alt_header="Do the userName and email.value fields have to point to the same value in SCIM?">

Yes. Both must match the email address the user uses to sign in to <Constant name="dbt_platform" /> (email value on the user object in your IdP that's used to sign in). If they don't match, a validation error will occur during provisioning.

</Expandable>

<Expandable alt_header="What is the recommended workflow for onboarding new users through SCIM?">

#### Entra ID

> Once SCIM provisioning is configured in Entra according to the schema requirements, provisioning begins automatically. Entra will sync users and groups assigned to the Entra app with <Constant name="dbt_platform" />.

Reviewer: ```suggestion
Once SCIM provisioning is configured in Entra ID according to the schema requirements, provisioning begins automatically. Entra ID will sync users and groups assigned to the Entra app with <Constant name="dbt_platform" />.
```

Reviewer: Hiya Mirna

Just amended this to match the standardisation of Entra ID for consistency and accuracy.

**For users:**
- Matching is based on the `userName` (and `email.value`) field, which should be set to the user's email — the same email used in <Constant name="dbt_platform" /> and expected at sign-in.
- If a user with that email already exists in <Constant name="dbt_platform" />, they will be linked and become SCIM-managed.
- If no matching user exists, a new user will be provisioned and invited through email.

**For groups:**
- Matching is based on the group name.
- If a group with the same name exists in <Constant name="dbt_platform" />, it will be linked and become SCIM-managed.
- If no match exists, a new group will be created through SCIM.

**Important considerations:**

> - Entra does not support importing existing users or groups into the app for SCIM — users and groups must already exist in Entra and be assigned to the app.

Reviewer: ```suggestion
- Entra ID does not support importing existing users or groups into the app for SCIM — users and groups must already exist in Entra ID and be assigned to the app.
```

Reviewer: same here

- To ensure users and groups become SCIM-managed, they must exist in both Entra and <Constant name="dbt_platform" /> with matching identifiers (email for users, name for groups).
- After setup, ongoing syncs automatically provision and manage any newly assigned users or groups in Entra.

#### Okta

> Unlike Entra, Okta does not perform continuous full syncs. Provisioning is triggered manually for the initial setup and then continues incrementally as changes are made.

Reviewer: ```suggestion
Unlike Entra ID, Okta does not run continuous full syncs. You trigger initial provisioning manually, then Okta continues provisioning incrementally as changes are made.
```

> When SCIM is enabled on an existing Okta SSO application, trigger the initial sync (if there are existing users in the Okta app) using the **Provision Users** button in the Okta Assignments tab. This performs a one-time sync of users to <Constant name="dbt_platform" />. Okta doesn't provide a native way to re-run this full sync &mdash; which means reprocessing typically requires removing and re-adding users.

Reviewer: ```suggestion
When you enable SCIM on an existing Okta SSO app, trigger the initial sync (if there are existing users in the Okta app) using the **Provision Users** button in the Okta Assignments tab. This runs a one-time sync of users to <Constant name="dbt_platform" />. Okta doesn't provide a native way to re-run this full sync &mdash; which means reprocessing typically requires removing and re-adding users.
```

> User matching is based on the `userName` (email) field, which must match the email used in <Constant name="dbt_platform" />. If a match is found, the user is linked and becomes SCIM-managed. If not, a new user is provisioned. If users exist in <Constant name="dbt_platform" /> but not in Okta, you can import them into Okta to keep both systems aligned.

Reviewer: Hiya Mirna, 

For visuals, this might look nice as bullets / replicating the earlier section - ie:

**For users:**
User matching is based on the `userName` (email) field, which must match the email used in <Constant name="dbt_platform" />:
- If a match is found, Okta links the user and marks them as SCIM-managed.
- If no match is found, Okta provisions a new user.
- If users exist in <Constant name="dbt_platform" /> but not in Okta, you can import them into Okta to keep both systems aligned.

Author: love this sugg!

After the initial sync, new users assigned to the Okta app are automatically provisioned into <Constant name="dbt_platform" />. Group memberships can then be managed through Okta's Push Groups feature, which allows groups and their memberships to be pushed into <Constant name="dbt_platform" />.

> 

Reviewer: Same as above - ie:

**For groups:**
- After initial user sync, assign new users to the Okta app to provision them automatically into <Constant name="dbt_platform" />.
- Manage group memberships with Okta Push Groups, which pushes groups and memberships into <Constant name="dbt_platform" />.

For new Okta SSO applications with no assigned users, either manually assign users to the app to trigger provisioning, or import users from <Constant name="dbt_platform" /> into Okta first. Users must exist in both systems with matching emails to link correctly and avoid duplicates.

For larger rollouts, consider working with your IdP admin to plan based on your setup and [SCIM license mapping](/docs/cloud/manage-access/scim-manage-user-licenses) to reduce manual steps.

</Expandable>

<Expandable alt_header="Do SSO group mappings still apply when SCIM is enabled?">

> No. SSO group mappings are bypassed for SCIM-managed users. Group membership is controlled entirely by your IdP.

Reviewer: ```suggestion
No. SSO group mappings do not apply to SCIM-managed users. For SCIM-managed users, group membership is controlled by your IdP.
```

Reviewer: Hiya Mirna

Could also add a link to the SSO mappings page

SSO group mappings only apply to users who authenticate with SSO and are not SCIM-managed.

This means that if you have a dbt group with SSO mappings, those mappings will not be applied to users who have been provisioned through SCIM.

</Expandable>

> <Expandable alt_header='What does "Allow manual updates" mean?'>

Reviewer: Hiya Mirna, 

This hasn't caused any issues in the preview but this alt_header doesn't use double quotes in alt_header like the others. Maybe amend it to match surrounding MDX style. Or is this because of the question / header being part quoted?

Author: hey @nataliefiann good quesiton! when i created the expandable component, i wanted to make sure it accepted single or double quotes. since this is a react component, it will accept both single and double quotes  - no need to add double quotes. did you get an error though?

> The **Allow manual updates** toggle controls whether an admin can manually update SCIM-managed entities, including the ability to send invites.

Reviewer: ```suggestion
The **Allow manual updates** toggle determines whether admins can manually modify SCIM-managed users and groups in <Constant name="dbt_platform" />, including sending invites.

```

> - **Disabled (default):** All user and group management is deferred entirely to your IdP. Manual changes in <Constant name="dbt_platform" /> to SCIM-managed users are blocked. This is the recommended setting, as any manual changes made while enabled can be overridden by subsequent SCIM requests.

Reviewer: This is probs just me, but I feel like this phrasing reads a bit awkward, maybe something like:

**Disabled (default):** Your IdP remains the source of truth for SCIM-managed users and groups. Manual changes to SCIM-managed users in <Constant name="dbt_platform" /> are blocked. This is the recommended setting because any manual changes made while enabled can be overwritten by later SCIM updates.

- **Enabled:** Admins can make manual changes to users in <Constant name="dbt_platform" /> alongside SCIM. This can be useful during initial setup and testing, but manual changes do not prevent SCIM from overriding them.

</Expandable>

> <Expandable alt_header="What happens to existing users and groups when I enable SCIM?">

Reviewer: Hiya Mirna

The toggle asks "What happens to existing users and groups..." but the body only mentions users. It might be worthwhile either adding a sentence about groups or amending the question.

Author: good shout - i'm going to wait to see what Allie says about the confirmation though as this is one of the questions that needs eng confirmation

Enabling SCIM does not automatically convert existing users and groups to SCIM-managed status. Your IdP only manages users who have been explicitly assigned to the app and provisioned through SCIM.

To bring existing users under SCIM management, assign them to the <Constant name="dbt_platform" /> app in your IdP and trigger a sync. Until a user is provisioned through SCIM, they remain unmanaged and are unaffected by SCIM sync operations.

</Expandable>

<Expandable alt_header="What happens when a user's email address changes in my IdP when SCIM is enabled?">

When a SCIM-managed user's email is updated in the IdP, <Constant name="dbt_platform" /> receives a SCIM request to update it. A confirmation email is sent to the new address, and once accepted, the change takes effect in <Constant name="dbt_platform" />.

</Expandable>

<Expandable alt_header="Does SCIM support automatic license assignment?">

> SCIM license mapping is supported for Okta. It is not supported for Microsoft Entra ID, though SSO license mapping is available as an alternative.

Reviewer: ```suggestion
SCIM license mapping is supported for Okta. It is not supported for Microsoft Entra ID. For Entra ID, use [SSO license mapping](/docs/cloud/manage-access/seats-and-users#mapped-configuration)
```

> For Okta license mapping setup, see [Manage user licenses with SCIM](/docs/cloud/manage-access/scim-manage-user-licenses).

Reviewer: ```suggestion
For Okta license mapping setup, refer to [Manage user licenses with SCIM](/docs/cloud/manage-access/scim-manage-user-licenses).
```

</Expandable>

<Expandable alt_header="Existing users and groups are not becoming SCIM-managed after enabling SCIM">

After enabling SCIM and completing the initial sync, pre-existing <Constant name="dbt_platform" /> users and groups do not show as SCIM-managed.

**Why this happens:** SCIM provisioning links an IdP identity to a <Constant name="dbt_platform" /> user record. Users created before SCIM was enabled don't have this link unless the IdP explicitly provisions them through SCIM.

**Steps to resolve:**

1. In your IdP, assign existing users to the <Constant name="dbt_platform" /> SCIM application.
2. Trigger a provisioning sync. The IdP will attempt to match users by their `userName` (typically their email address) and establish the SCIM-managed link.
3. For Entra ID, note that the provisioning sync is one-way (push only) — there is no import option to pull existing dbt users into Entra as a managed identity.
4. If users aren't being matched correctly after a sync, confirm that the `userName` attribute in your IdP exactly matches the email on the user's <Constant name="dbt_platform" /> account, including case.

> 

Reviewer: Hiya Mirna

As a suggestion to enforce the to resolve this issue, maybe this could we make this section provider-neutral and move the Entra-specific point into a ‘Provider note’? That keeps the troubleshooting steps reusable for both Okta and Entra?

Author: i can try but it also can interrupt the flow if they are entra and trying to import in the midst

Reviewer: That makes sense — I agree we don’t want to interrupt the flow for Entra users while they’re in the middle of setup.

Maybe something visually to distinguish it (like a callout or note block) so it doesn’t break the flow but is still easy to spot. For example:

**Steps to resolve:**

1. In your IdP, assign existing users and groups to the <Constant name="dbt_platform" /> SCIM application.
2. Trigger a provisioning sync to create SCIM-managed links for matched records.

:::info Entra ID note
Provisioning in Entra ID is push-only. You cannot import existing users from <Constant name="dbt_platform" /> into Entra ID.
:::

3. Verify matching attributes:
   - Users: `userName` (typically email) matches the email in <Constant name="dbt_platform" />.
   - Groups: group names match between your IdP and <Constant name="dbt_platform" />.
4. If matching still fails, review provisioning logs in your IdP and retry.

</Expandable>

- If no match exists, a new group will be created through SCIM.

**Important considerations:**
- Entra ID does not support importing existing users or groups into the app for SCIM — users and groups must already exist in Entra ID and be assigned to the app.
- To ensure users and groups become SCIM-managed, they must exist in both Entra and <Constant name="dbt_platform" /> with matching identifiers (email for users, name for groups).
- After setup, ongoing syncs automatically provision and manage any newly assigned users or groups in Entra.

#### Okta

Unlike Entra ID, Okta does not run continuous full syncs. You trigger initial provisioning manually, then Okta continues provisioning incrementally as changes are made.

When you enable SCIM on an existing Okta SSO app, trigger the initial sync (if there are existing users in the Okta app) using the **Provision Users** button in the Okta Assignments tab. This runs a one-time sync of users to <Constant name="dbt_platform" />. Okta doesn't provide a native way to re-run this full sync &mdash; which means reprocessing typically requires removing and re-adding users.

> User matching is based on the `userName` (email) field, which must match the email used in <Constant name="dbt_platform" />. If a match is found, the user is linked and becomes SCIM-managed. If not, a new user is provisioned. If users exist in <Constant name="dbt_platform" /> but not in Okta, you can import them into Okta to keep both systems aligned.

Author: ```suggestion
**For users:**
User matching is based on the `userName` (email) field, which must match the email used in <Constant name="dbt_platform"/>:

- If a match is found, Okta links the user and marks them as SCIM-managed.
- If no match is found, Okta provisions a new user.
- If users exist in <Constant name="dbt_platform"/> but not in Okta, you can import them into Okta to keep both systems aligned.
```

> After the initial sync, new users assigned to the Okta app are automatically provisioned into <Constant name="dbt_platform" />. Group memberships can then be managed through Okta's Push Groups feature, which allows groups and their memberships to be pushed into <Constant name="dbt_platform" />.

Author: ```suggestion
**For groups:**
- After initial user sync, assign new users to the Okta app to provision them automatically into <Constant name="dbt_platform" />.
- Manage group memberships with Okta's Push Groups feature, which pushes groups and memberships into <Constant name="dbt_platform" />.
```

For new Okta SSO applications with no assigned users, either manually assign users to the app to trigger provisioning, or import users from <Constant name="dbt_platform" /> into Okta first. Users must exist in both systems with matching emails to link correctly and avoid duplicates.

For larger rollouts, consider working with your IdP admin to plan based on your setup and [SCIM license mapping](/docs/cloud/manage-access/scim-manage-user-licenses) to reduce manual steps.

</Expandable>

<Expandable alt_header="Do SSO group mappings still apply when SCIM is enabled?">

No. SSO group mappings do not apply to SCIM-managed users. For SCIM-managed users, group membership is controlled by your IdP.

SSO group mappings only apply to users who authenticate with SSO and are not SCIM-managed.

This means that if you have a dbt group with SSO mappings, those mappings will not be applied to users who have been provisioned through SCIM.

</Expandable>

<Expandable alt_header='What does "Allow manual updates" mean?'>

The **Allow manual updates** toggle determines whether admins can manually modify SCIM-managed users and groups in <Constant name="dbt_platform" />, including sending invites.

> - **Disabled (default):** All user and group management is deferred entirely to your IdP. Manual changes in <Constant name="dbt_platform" /> to SCIM-managed users are blocked. This is the recommended setting, as any manual changes made while enabled can be overridden by subsequent SCIM requests.

Author: ```suggestion
- **Disabled (default)**: Your IdP remains the source of truth for SCIM-managed users and groups. Manual changes to SCIM-managed users in <Constant name="dbt_platform"/> are blocked. This is the recommended setting because any manual changes made while enabled can be overwritten by later SCIM updates.
```

For new Okta SSO applications with no assigned users, you can either manually assign users to the app (which will trigger provisioning), or import users from <Constant name="dbt_platform" /> into Okta before syncing. Ensuring that users exist in both systems and that their emails match is critical for proper linking and avoiding duplicate user creation.

For larger rollouts, consider working with your IdP admin to plan based on your setup and [SCIM license mapping](/docs/cloud/manage-access/scim-manage-user-licenses) to reduce manual steps.

</Expandable>

<Expandable alt_header="Do SSO group mappings still apply when SCIM is enabled?">

No. For users who are provisioned and managed through SCIM, SSO group mappings are bypassed entirely. Group membership for SCIM-managed users is controlled by your IdP. SSO group mappings only apply to users who authenticate through SSO and are **not** SCIM-managed.

This means that if you have a dbt group with SSO mappings, those mappings will not be applied to users who have been provisioned through SCIM.

</Expandable>

> {/* TODO: Needs engineering verification before publishing — confirm the toggle behavior description is accurate, especially around the "Enabled" state partial sentence in the source */}

Author: ```suggestion
```

<Expandable alt_header='What does "Allow manual updates" mean?'>

The **Allow manual updates** toggle controls whether an admin can manually update SCIM-managed entities, including the ability to send invites.

- **Disabled (default):** All user and group management is deferred entirely to your IdP. Manual changes in <Constant name="dbt_platform" /> to SCIM-managed users are blocked. This is the recommended setting, as any manual changes made while enabled can be overridden by subsequent SCIM requests.
- **Enabled:** Admins can make manual changes to users in <Constant name="dbt_platform" /> alongside SCIM. This can be useful during initial setup and testing, but manual changes do not prevent SCIM from overriding them.

</Expandable>

{/* TODO: Needs engineering verification before publishing — confirm behavior across both Okta and Entra and that the Entra "push only" note is still accurate */}

> <Expandable alt_header="What happens to existing users and groups when I enable SCIM?">

Author: flag to confirm

Existing users and groups in <Constant name="dbt_platform" /> are **not** automatically converted to SCIM-managed status when you first enable SCIM. Your IdP will only manage users that have been explicitly assigned to the <Constant name="dbt_platform" /> application in their IdP and provisioned with SCIM.

To bring existing users under SCIM management, assign them to the <Constant name="dbt_platform" /> app in your IdP and trigger a sync. Until a user is provisioned with SCIM, they remain unmanaged and are unaffected by SCIM sync operations.

</Expandable>

{/* TODO: Needs engineering verification before publishing — confirm the email confirmation flow described is current */}

> <Expandable alt_header="What happens when a user's email address changes in my IdP when SCIM is enabled?">

Author: flag to confirm

Reviewer 2: Sounds good to me :)

If the user is SCIM-managed, when their email is updated in the IdP, <Constant name="dbt_platform" /> will receive a request from SCIM to update their email. An email will be sent to the new address to confirm the change. Once accepted, the user's email will be officially updated in <Constant name="dbt_platform" />.

</Expandable>

{/* TODO: Needs engineering verification before publishing — the Entra ID license mapping sentence is incomplete in the source. Confirm whether Entra supports license mapping at all, or only via SSO license mapping. Add correct links once confirmed. */}

> <Expandable alt_header="Does SCIM support automatic license assignment?">

Author: flag to confirm

Reviewer 2: Looks good :)

SCIM license mapping is supported for **Okta**. It is not supported for **Microsoft Entra ID** — however, SSO license mapping is supported for Entra ID and can be configured that way.

For Okta license mapping setup, see [Manage user licenses with SCIM](/docs/cloud/manage-access/scim-manage-user-licenses).

</Expandable>

{/* TODO: Needs engineering sign-off — DO NOT publish until confirmed. Answer is inferred from system behavior, not explicitly confirmed by engineering. See also: SSO FAQ for the same question. */}

> <Expandable alt_header="Can I use Okta for SSO and Entra ID for SCIM (or vice versa)?">

Author: flag to confirm

Reviewer 2: looks good :)

</Expandable>

---

## Troubleshooting

<Expandable alt_header='"All users must have licenses on the account" error'>

This error occurs when a SCIM group push includes a user who has not yet been licensed in <Constant name="dbt_platform" /> — typically because the user hasn't accepted their invitation yet.

**Steps to resolve:**

1. Identify the user(s) causing the error from your IdP's provisioning logs.
2. Check whether those users have accepted their <Constant name="dbt_platform" /> invitation. Users are not licensed until they complete this step.
3. Once the user accepts their invitation and logs in, retry the group push from your IdP.
4. If the invitation has expired, remove the user from the push group temporarily, re-invite them using <Constant name="dbt_platform" />, have them accept, and then re-add them to the push group.

</Expandable>

{/* TODO: Needs engineering verification before publishing — confirm the Entra "push only" behavior is still accurate and that there is no import option */}

> <Expandable alt_header="Existing users and groups are not becoming SCIM-managed after enabling SCIM">

Author: flag to confirm

**Why this happens:** SCIM provisioning creates a managed association between an IdP identity and a <Constant name="dbt_platform" /> user record. Users created before SCIM was enabled do not have this association unless the IdP explicitly provisions them through SCIM.

**Steps to resolve:**

1. In your IdP, assign existing users to the <Constant name="dbt_platform" /> SCIM application.
2. Trigger a provisioning sync. The IdP will attempt to match existing users by their `userName` (typically email address) and establish the SCIM-managed link.
3. For Entra ID, note that the provisioning sync is one-way (push only) — there is no import option to pull existing dbt users into Entra as a managed identity.
4. If users are not being matched correctly after a sync, confirm that the `userName` attribute in your IdP matches the email address on the user's <Constant name="dbt_platform" /> account exactly, including case.

</Expandable>

{/* TODO: Needs engineering verification before publishing — confirm the AzureActiveDirectory service tag approach is still current and that the Admin API method is supported */}

> <Expandable alt_header="Azure SCIM provisioning fails due to IP allowlisting">

Author: flag to confirm

Reviewer 2: I might say Entra to stay consistent

---
title: "SCIM FAQs and troubleshooting"
description: "Common questions and troubleshooting for SCIM provisioning in dbt platform"
id: "scim-faq"
sidebar: "SCIM FAQ and troubleshooting"
---

# SCIM FAQs and troubleshooting <Lifecycle status="managed, managed_plus" />

Find answers to common questions about configuring and using SCIM provisioning in <Constant name="dbt_platform" />, plus guidance for resolving common issues.

## FAQs

> <Expandable alt_header="Do the userName and email.value fields have to point to the same value in SCIM?">

Reviewer 2: I might update "have to point to" --> "have to be"

</Expandable>

<Expandable alt_header="Does SCIM support automatic license assignment?">

SCIM license mapping is supported for Okta. It is not supported for Microsoft Entra ID. For Entra ID, use [SSO license mapping](/docs/cloud/manage-access/seats-and-users#mapped-configuration)

For Okta license mapping setup, refer to [Manage user licenses with SCIM](/docs/cloud/manage-access/scim-manage-user-licenses).

</Expandable>

<Expandable alt_header="Can I use Okta for SSO and Entra ID for SCIM (or vice versa)?">

SSO and SCIM should be configured using the same IdP. Using different providers — for example, Okta for SSO and Entra ID for SCIM &mdash; can cause discrepancies in user state and unintended behavior.

If your organization uses separate IdPs for authentication and directory management, contact your account team to discuss your options.

</Expandable>

---

## Troubleshooting

<Expandable alt_header='"All users must have licenses on the account" error'>

> This error occurs when a SCIM group push includes a user who hasn't been licensed in <Constant name="dbt_platform" /> yet &mdash; typically because they haven't accepted their invitation.

Reviewer 2: This isn't correct anymore. It only happens now when a user does not have a license on the account. The steps to fix, are to ensure everyone assigned to the group has an active license on the account

3. Once the user accepts and signs in, retry the group push from your IdP.
4. If the invitation has expired, remove the user from the push group temporarily, re-invite them using <Constant name="dbt_platform" />, have them accept, then re-add them to the group.

</Expandable>

<Expandable alt_header="Existing users and groups are not becoming SCIM-managed after enabling SCIM">

After enabling SCIM and completing the initial sync, pre-existing <Constant name="dbt_platform" /> users and groups do not show as SCIM-managed.

**Why this happens:** SCIM provisioning links an IdP identity to a <Constant name="dbt_platform" /> user record. Users created before SCIM was enabled don't have this link unless the IdP explicitly provisions them through SCIM.

**Steps to resolve:**

1. In your IdP, assign existing users to the <Constant name="dbt_platform" /> SCIM application.
2. Trigger a provisioning sync. The IdP will attempt to match users by their `userName` (typically their email address) and establish the SCIM-managed link.
3. For Entra ID, note that the provisioning sync is one-way (push only) — there is no import option to pull existing dbt users into Entra as a managed identity.

> 4. If users aren't being matched correctly after a sync, confirm that the `userName` attribute in your IdP exactly matches the email on the user's <Constant name="dbt_platform" /> account, including case.

Reviewer 2: and the `email.value` attribute

</Expandable>

<Expandable alt_header="Azure SCIM provisioning fails due to IP allowlisting">

> If your <Constant name="dbt_platform" /> account has **IP restrictions** enabled, Azure's SCIM provisioning requests may be blocked. Azure's provisioning service IPs rotate approximately every two weeks and can't be statically allowlisted.

Reviewer 2: same here: Entra

Author: ugh sorry what a miss!
